tests/php/src/Optimizer/HeroCandidateFilteringTest.php

```diff
@@ -0,0 +1,175 @@
+		$this->assertSame(
+			'<img data-hero-candidate src="https://example.com/header.jpg">',
+			$this->instance->filter_header_image_tag(
+				'<img src="https://example.com/header.jpg">'
+			)
+		);
+
+		$this->assertSame(
+			'<div>Nothing</div>',
+			$this->instance->filter_header_image_tag(
+				'<div>Nothing</div>'
+			)
+		);
+	}
+
+	/** @covers ::filter_attachment_image_attributes()  */
+	public function test_filter_attachment_image_attributes() {
+		$attachment_id = self::factory()->attachment->create_upload_object( DIR_TESTDATA . '/images/canola.jpg', 0 );
+		$attachment    = get_post( $attachment_id );
+		$post_ids      = self::factory()->post->create_many(
+			2,
+			[ 'post_content' => __FUNCTION__ ]
+		);
+
+		$search_request_uri = sprintf( '/?s=%s&orderby=ID&order=asc', __FUNCTION__ );
+
+		$initial_attrs = [ 'data-foo' => 'bar' ];
+		$merged_attrs  = array_merge(
+			$initial_attrs,
+			self::ARRAY_WITH_DATA_HERO_CANDIDATE_ATTR
+		);
+
+		// Make sure attachment doesn't get data-hero-candidate when it wasn't assigned as a featured image.
+		$this->go_to( $search_request_uri );
+		$this->assertTrue( is_search() );
+		$this->assertSame(
+			$post_ids,
+			wp_list_pluck( $GLOBALS['wp_query']->posts, 'ID' ),
+			'Expected query to match the posts created.'
+		);
+		$this->assertEquals(
+			$initial_attrs,
+			$this->instance->filter_attachment_image_attributes( $initial_attrs, $attachment )
+		);
+
+		// When attachment is featured image of second post, it still does not get the attribute.
+		$this->go_to( $search_request_uri );
+		set_post_thumbnail( $post_ids[1], $attachment->ID );
+		$this->assertEquals(
+			$initial_attrs,
+			$this->instance->filter_attachment_image_attributes( $initial_attrs, $attachment )
+		);
+
+		// When attachment is featured image of first post, it does get the attribute.
+		delete_post_thumbnail( $post_ids[1] );
+		set_post_thumbnail( $post_ids[0], $attachment->ID );
+		$this->go_to( $search_request_uri );
+		$this->assertEquals(
+			$merged_attrs,
+			$this->instance->filter_attachment_image_attributes( $initial_attrs, $attachment )
+		);
+
+		// When on the singular post without a featured image, no attribute is added.
+		delete_post_thumbnail( $post_ids[0] );
+		set_post_thumbnail( $post_ids[1], $attachment->ID );
+		$this->go_to( get_permalink( $post_ids[0] ) );
+		$this->assertEquals(
+			$initial_attrs,
+			$this->instance->filter_attachment_image_attributes( $initial_attrs, $attachment )
+		);
+
+		// When on the singular post has a featured image, an attribute is added.
+		delete_post_thumbnail( $post_ids[1] );
+		set_post_thumbnail( $post_ids[0], $attachment->ID );
+		$this->go_to( get_permalink( $post_ids[0] ) );
+		$this->assertEquals(
+			$merged_attrs,
+			$this->instance->filter_attachment_image_attributes( $initial_attrs, $attachment )
+		);
+	}
+
+	/** @covers ::add_data_hero_candidate_attribute()  */
+	public function add_data_hero_candidate_attribute() {
```

Comment:
This test was not being executed.

```suggestion
	public function test_add_data_hero_candidate_attribute() {
```

Reply:
Oops! Committed your suggestion in this commit: 67a2f91d6468195f8ee3f056efb73a67032ed06e